sound/soc/sof/ipc4-topology.c

```diff
@@ -2075,6 +2092,10 @@ static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
 		return output_fmt_index;
 	}
 
+	/* adjust the input/output buffer size based on the period size set in topology */
+	process->base_config.ibs *= process->buffer_period_size;
+	process->base_config.obs *= process->buffer_period_size;
```

Comment:
is this correct to multiply the pin_format->buffer_size and ibs/obs? Isn't the pin_format->buffer_size derived from IBS/OBS?

Looks like a double multiplication to me...

Reply:
Hi, @RanderWang , could you help to double check above? sorry I am not really familiar this part.

Reply:
this change should be made in topology since driver just pass it to fw.
please check audio_format.conf. you can override it for DP module
````c
      # math expression for computing input/put buffer sizes
        # for 11.025 22.05, 44.1, 88.2 and 176.4khz, we need to round it to ceiling value
        ibs "$[($in_channels * ($[($in_rate + 999)] / 1000)) * ($in_bit_depth / 8)]"
        obs "$[($out_channels * ($[($out_rate + 999)] / 1000)) * ($out_bit_depth / 8)]"
````

Reply:
And if the process->buffer_period_size is used to build IBS or OBS only, we can do it in topology and no need to change driver ?

Reply:
Thanks @RanderWang , updated the https://github.com/thesofproject/sof/pull/8403, hope to have same net results :)

Reply:
This PR can be closed then?